csp_billing_adapter_local/plugin.py

```diff
@@ -13,66 +13,102 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+"""
+plugin.py is part of csp-billing-adapter-local and provides the local storage
+plugin
+"""
 
 import json
-import os
+import logging
+
+from json.decoder import JSONDecodeError
+from pathlib import Path
 
 import csp_billing_adapter
 
 from csp_billing_adapter.config import Config
 
-cache_file = os.path.expanduser('~/csp-adapter/cache.json')
-csp_config_file = os.path.expanduser('~/csp-adapter/csp-config.json')
+log = logging.getLogger('CSPBillingAdapter')
+
+LOCAL_DIR = '.local'
```

Comment:
I suppose `local` was chosen for the use case when the adapter does not run as part of a cluster. As such this would imply running as a daemon or container in a VM. In either case we should stick to accepted and FHS (Filesystem Hierarchy Standard) locations. Cache files got to `/var/cache`

Reply:
so there are two files that get saved in the local and  memory-cache plugins.  One is the adapter state csp-config.json and the other is the usage records cache.  do we want these both in /var/cache/csp-billing-adapter?  Should we update the name of csp-config.json  per your other comment to csp-state.json or such?

Reply:
I would consider the usage records to be a cache, as that will get updated hourly. That said the specification also states an application should be able to re-generate data that is in `/var/cache`, which is not true in our case. The file with the information the adapter makes available to the application should be in `/var/lib`, given the re-generation problem with the usage record we should also keep it in `/var/lib`